Source/WebCore/Modules/WebGPU/GPUBuffer.cpp

```diff
@@ -47,12 +53,15 @@ void GPUBuffer::mapAsync(GPUMapModeFlags mode, std::optional<GPUSize64> offset,
 
 Ref<JSC::ArrayBuffer> GPUBuffer::getMappedRange(std::optional<GPUSize64> offset, std::optional<GPUSize64> size)
 {
-    auto mappedRange = m_backing->getMappedRange(offset.value_or(0), size);
-    return ArrayBuffer::create(mappedRange.source, mappedRange.byteLength);
+    m_mappedRange = m_backing->getMappedRange(offset.value_or(0), size);
```

Comment:
Do we need to validate that size is <= the size of the buffer?

Reply:
This is validated in WebGPU.framework layer as getMappedRange could be called by the native client.

Do you think we should validate in the web process layer too? It would be double validation and does not seem to occur for the other calls in WebCore from what I can tell

Reply:
Maybe put a comment in about this, in case our Security folks question it when they do a code inspection?

Reply:
Hmm... I suck at github. That comment is marked as Pending, but it is just a suggestion.

Reply:
I added a comment :+1: